python/ray/experimental/sgd/pytorch/pytorch_trainer.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import sys
+import torch
+
+import ray
+
+from ray.experimental.sgd.pytorch.pytorch_runner import PyTorchRunner
+from ray.experimental.sgd.pytorch import utils
+
+
+class PyTorchTrainer(object):
+    """Train a PyTorch model using distributed PyTorch.
+
+    Launches a set of actors which connect via distributed PyTorch and
+    coordinate gradient updates to train the provided model.
+    """
+
+    def __init__(
+            self,
+            model_creator,
+            data_creator,
+            optimizer_creator=None,  # SGD as default TODO: handle this
+            config=None,
+            num_replicas=1,
+            resources_per_replica=None,
+            batch_size=16,
+            backend="gloo"):
+        """Sets up the PyTorch trainer.
+
+        Args:
+            model_creator (dict -> torch.nn.Module): creates the model
+                using the config.
+            data_creator (dict -> Dataset, Dataset): creates the training
+                and validation data sets using the config.
+            optimizer_creator (model, dict -> loss, optimizer): creates the
+                loss and optimizer using the config.
+            config (dict): configuration passed to 'model_creator',
+                'data_creator', and 'optimizer_creator'.
+            batch_size (int): batch size used for SGD.
```

Comment:
This is misleading (as is "batch_size" for runner) - can you clarify (i.e., per device, or per update)?